StateWrappers.as

```diff
@@ -1,52 +1,99 @@
 #if MP4
 
+const uint WheelsStartOffset = 0x53C;
+const uint WheelStructLength = 0x24;
+
+namespace VehicleState
+{
+	shared enum EffectFlags {
+		FreeWheeling = 1,
+		ForcedAcceleration = 2,
+		NoBrakes = 4,
+		NoSteering = 8,
+		NoGrip = 16
+	}
+}
+
 shared class CSceneVehicleVisState
 {
-	CTrackManiaPlayer@ m_player;
-	CTrackManiaScriptPlayer@ m_scriptapi;
+	CMwNod@ m_vis;
 
-	CSceneVehicleVisState(CTrackManiaPlayer@ player)
+	CSceneVehicleVisState(CMwNod@ m_vis)
 	{
-		@m_player = player;
-		@m_scriptapi = m_player.ScriptAPI;
+		@this.m_vis = m_vis;
 	}
 
-	float get_InputSteer() { if (m_scriptapi is null) { return 0; } return m_scriptapi.InputSteer; }
-	float get_InputGasPedal() { if (m_scriptapi is null) { return 0; } return m_scriptapi.InputGasPedal; }
-	float get_InputBrakePedal() { if (m_scriptapi is null) { return 0; } return m_scriptapi.InputIsBraking ? 1 : 0; }
-	bool get_InputIsBraking() { if (m_scriptapi is null) { return false; } return m_scriptapi.InputIsBraking; }
+	CSceneVehicleVisState@ get_AsyncState()
+	{
+		return this;
+	}
```

Comment:
Do we need this?

Reply:
Oh, you explained this in the PR itself. I think this is okay to keep, but maybe put a comment explaining what it's actually there for? (eg. "Compatibility with the Next API" or something along those lines)